examples/data-solutions/dp-foundation/03-orc-composer.tf

```diff
@@ -0,0 +1,133 @@
+      subnetwork      = module.orc-vpc[0].subnet_self_links["${var.composer_config.region}/${local.prefix_orc}-subnet"]
+      tags            = ["composer-worker", "http-server", "https-server"]
+      ip_allocation_policy {
+        use_ip_aliases                = "true"
+        cluster_secondary_range_name  = "pods"
+        services_secondary_range_name = "services"
+      }
+    }
+    software_config {
+      image_version = "composer-1.17.5-airflow-2.1.4"
+      env_variables = {
+        DTL_EXP_PRJ        = module.dtl-0-prj.project_id
+        DTL_EXP_BQ_DATASET = module.dtl-exp-bq-0.dataset_id
+        DTL_EXP_GCS        = module.dtl-exp-cs-0.url
+        DTL_L0_PRJ         = module.dtl-0-prj.project_id
+        DTL_L0_BQ_DATASET  = module.dtl-0-bq-0.dataset_id
+        DTL_L0_GCS         = module.dtl-0-cs-0.url
+        DTL_L1_PRJ         = module.dtl-1-prj.project_id
+        DTL_L1_BQ_DATASET  = module.dtl-1-bq-0.dataset_id
+        DTL_L1_GCS         = module.dtl-1-cs-0.url
+        DTL_L2_PRJ         = module.dtl-2-prj.project_id
+        DTL_L2_BQ_DATASET  = module.dtl-2-bq-0.dataset_id
+        DTL_L2_GCS         = module.dtl-2-cs-0.url
+        GCP_REGION         = var.composer_config.region
+        LND_PRJ            = module.lnd-prj.project_id
+        LND_BQ             = module.lnd-bq-0.dataset_id
+        LND_GCS            = module.lnd-cs-0.url
+        LND_PS             = module.lnd-ps-0.id
+        LOD_PRJ            = module.lod-prj.project_id
+        LOD_GCS_STAGING    = module.lod-cs-df-0.url
+        LOD_NET_VPC        = module.lod-vpc[0].self_link
+        LOD_NET_SUBNET     = module.lod-vpc[0].subnet_self_links["${var.composer_config.region}/${local.prefix_lod}-subnet"]
+        LOD_SA_DF          = module.lod-sa-df-0.email
+        ORC_PRJ            = module.orc-prj.project_id
+        ORC_GCS            = module.orc-cs-0.url
+        TRF_PRJ            = module.trf-prj.project_id
+        TRF_GCS_STAGING    = module.trf-cs-df-0.url
+        TRF_SA_DF          = module.trf-sa-df-0.email
+        TRF_SA_BQ          = module.trf-sa-bq-0.email
+      }
+    }
+    private_environment_config {
+      enable_private_endpoint    = "true"
+      master_ipv4_cidr_block     = var.composer_config.ip_range_gke_master
+      cloud_sql_ipv4_cidr_block  = var.composer_config.ip_range_cloudsql
+      web_server_ipv4_cidr_block = var.composer_config.ip_range_web_server
+    }
+
+    dynamic "encryption_config" {
+      for_each = try(local.service_encryption_keys.composer != null, false) ? { 1 = 1 } : {}
+      content {
+        kms_key_name = try(local.service_encryption_keys.composer != null, false) ? try(local.service_encryption_keys.composer, null) : null
+      }
+    }
+
+    # web_server_network_access_control {
+    #   allowed_ip_range {
+    #     value       = "172.16.0.0/12"
+    #     description = "Allowed ip range"
+    #   }
+    # }
+  }
+  depends_on = [
```

Comment:
Do you really need this? Are there no implicit dependencies via references? What's the point of depending Composer from everything else?

Reply:
Since references are used to set Cloud Composer environment variables, it doesn't get implicit dependencies. Do you see a better way?

Reply:
The references in `software_config` should be enough, did you verify if they work?

Reply:
Did a new test now, it works. Not sure why it didn't work in the past.